test/facade_converter_test.ts

```diff
@@ -237,6 +243,99 @@ var y = x.length;`);
 }`);
   });
 
+  describe('promises', () => {
+    it('translates into Futures', () => {
+      expectWithTypes('let x: Promise = Promise.resolve(1);').to.equal(`import "dart:async";
+
+Future x = Future.value(1);`);
+      expectWithTypes('let x: Promise = Promise.reject(1);').to.equal(`import "dart:async";
+
+Future x = Future.error(1);`);
+      expectWithTypes('let x: Promise = new Promise((resolve) => {resolve(1);});')
+          .to.equal(`import "dart:async";
+
+Future x = (() {
+  Completer _completer$$ts2dart$0 = new Completer();
+  var resolve = _completer$$ts2dart$0.complete;
+  (() {
+    resolve(1);
+  })();
+  return _completer$$ts2dart$0.future;
+})();`);
+      expectWithTypes('let x: Promise = new Promise((resolve, reject) => {resolve(1);});')
+          .to.equal(`import "dart:async";
+
+Future x = (() {
+  Completer _completer$$ts2dart$0 = new Completer();
+  var resolve = _completer$$ts2dart$0.complete;
+  var reject = _completer$$ts2dart$0.completeError;
+  (() {
+    resolve(1);
+  })();
+  return _completer$$ts2dart$0.future;
+})();`);
```

Comment:
After review, without AST transform this will be impossible. The problem is that the code preceding the syntax emission point can be any value left hand side expression, so we can't change it or copy it properly.

The transform here would be simple that maybe it could be done in a pass; create the Completer before the Future declaration and change the value of the future by `completer.future`. This would remove the first function call.